File: 4. Week 4 - Pie Chart/exercise/r/visualization.R
# Visualize Time of Day Runs using ggplot2 (Pie Chart)

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

# We need to make sure the path points to our data file
data_path <- "../../data/time_of_day_counts.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Calculate percentages for labels if needed
  # df$percent <- df$Count / sum(df$Count) * 100
  # df$label <- paste0(df$Category, "\n", round(df$percent, 1), "%")
  
  # Plot
  # TODO: Create the pie chart using ggplot2
  # Note: A pie chart in ggplot2 is typically a bar chart with coord_polar
  
  # p <- ggplot(df, aes(x = "", y = Count, fill = Category)) +
  #   geom_bar(stat = "identity", width = 1) + ...
    
  
  # Uncomment to save
  # ggsave("time_of_day_pie.png", plot = p, width = 8, height = 8)
  # print("Chart saved to time_of_day_pie.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 4. Week 4 - Pie Chart/exercise/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/time_of_day_counts.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load data
    # df = ...
    
    # Plot
    plt.figure(figsize=(8, 8))
    
    # Colors (Purple Theme)
    colors = ['#D1C4E9', '#9575CD', '#673AB7', '#311B92']
    
    # TODO: Plot the pie chart
    # Hint: Use plt.pie(..., labels=..., autopct=..., colors=...)
    # plt.pie(...)
    
    # TODO: Add title
    # plt.title(...)
    
    plt.tight_layout()
    # TODO: Save the chart
    # plt.savefig(...)
    
    print("Chart saved to time_of_day_pie.png")
    plt.show()

if __name__ == "__main__":
    visualize()
